src/http.c

```diff
@@ -279,7 +279,17 @@ int mg_http_parse(const char *s, size_t len, struct mg_http_message *hm) {
   hm->uri.buf = (char *) s;
   while (s < end && (n = clen(s, end)) > 0) s += n, hm->uri.len += n;
   while (s < end && s[0] == ' ') s++;  // Skip spaces
+  is_response = mg_ncasecmp(hm->method.buf, "HTTP/", 5) == 0;
   if ((s = skiptorn(s, end, &hm->proto)) == NULL) return false;
+  if (!is_response && hm->proto.len != 0) {
+    if (hm->proto.len == 8 && mg_ncasecmp(hm->proto.buf, "HTTP/", 5) == 0) {
```

Comment:
You're doing mg_ncasecmp twice! I suggest to remove mg_ncasecmp() call on line 282, and refactor a condition on like 284 to this:

```c
// If we're given a version, check that it is HTTP/x.x
if (mg_ncasecmp(hm->proto.buf, "HTTP/", 5) == 0) &&
     hm->proto.len != 8 && 
      hm->proto.buf[6] != '.' &&
      (hm->proto.buf[5] < '0' || hm->proto.buf[5] > '9') &&  
      (hm->proto.buf[7] < '0' || hm->proto.buf[7] > '9') {
  return -1;
}
```

Reply:
It is not exactly used twice. First, I am using it to detect if the http packet is a response or a request (responses start with 'HTTP/' string). In the 2nd usage, I am checking the version (hm->proto.buf), which also happens to start with 'HTTP/'.